lib/tasks/test.rake

```diff
@@ -1,3 +1,13 @@
+class Rake::Task
+  alias_method :original_enhance, :enhance
+
+  def enhance(deps=nil, &block)
+    Foreman::Deprecation.deprecation_warning('1.15', 'a plugin enhanced rake task "test", use "test:plugins" instead') unless defined? @@warning_issued
```

Comment:
`::Foreman::Deprecation` isn't really defined at this point, it throws `UninitializedConstantError`

Aside from that, I'm not sure this is doing anything, a lot of tasks internally enhance other tasks, this catches any 'rake_task.enhance', not just 'test' enhancements.

Surprisingly enough I don't see `test:discovery` in the list of tasks, let me investigate a bit - to see the list of tasks that would call this, substitute this line by `puts self.name + ' enhanced rake task "test", use "test:plugins" instead'`